### PR TITLE
openssh: 7.9p1 -> 8.0p1

### PR DESCRIPTION
**Pull request by edef1c**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Release notes: https://www.openssh.com/releasenotes.html

File: pkgs/tools/networking/openssh/default.nix [hunk 1]
```diff
@@ -13,16 +13,16 @@ let
...
-    sha256 = "14j9cabb3gkhkjc641zbiv29mbvsmgsvis3fbj8ywsd21zc7m2wv";
+    sha256 = "1cnqlkrxzq7f2yy1iw9x0cp1w8g1skdc7q8r5x7s7chg3i0x5w92";
   };
 
 in
 with stdenv.lib;
 stdenv.mkDerivation rec {
-  pname = "openssh";
-  version = if hpnSupport then "7.8p1" else "7.9p1";
+  name = "openssh-${version}";
```
Comment (mmahut) [inline pkgs/tools/networking/openssh/default.nix]: What is the reason of moving away from ```pname```?

Comment (edef1c) [reply pkgs/tools/networking/openssh/default.nix]: Whoops. That was a rebase mistake, I'll fix that.

**Comment by dtzWill** [comment]
Dupe: #59806, but kinda shocked that's still in limbo.  Glad to have it pinged/pushed!

**Pull request closed without merging**